modules/javafx.controls/src/main/java/javafx/scene/control/TreeTableRow.java

```diff
@@ -441,7 +441,7 @@ private void updateSelection() {
         if (index == -1 || getTreeTableView() == null) return;
         if (getTreeTableView().getSelectionModel() == null) return;
 
-        boolean isSelected = getTreeTableView().getSelectionModel().isSelected(index);
+        boolean isSelected = getTreeTableView().getSelectionModel().isSelected(index, null);
         if (isSelected() == isSelected) return;
```

Comment:
good example of when using the two-param method makes sense - we probably should do the same in TableRow (which does some convoluted ifs for the same result) to keep both as similar as possible. Might be done here or in a new low-priority  follow-up issue

Reply:
unnecessary in the case of TableRow.
the change in TreeTableRow was necessary because otherwise selecting a cell by mouse selected the whole row.

Reply:
> unnecessary in the case of TableRow. 

that is .. why exactly?

> the change in TreeTableRow was necessary because otherwise selecting a cell by mouse selected the whole row.

it still does (if all cells are selected)

see me in grinning - my first guess was exactly the same as yours, but the moment I was about answering your comment (with "should behave the same and _they do_") some memory neuron fired and made me curious enough the check. The outcome is [JDK-8292353](https://bugs.openjdk.org/browse/JDK-8292353)

as that is unrelated to this fix (same before/after), we are fine here

Reply:
[JDK-8292353](https://bugs.openjdk.org/browse/JDK-8292353) can be trivially fixed (PR to be filed).